testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -389,55 +389,76 @@ public void testTrinoCreatesSparkDrops()
                 .containsOnly(row2);
 
-        onTrino().executeQuery("DROP TABLE " + trinoTableName);
+        Row row3 = row("a", new byte[]{15, -15, 2, -16, -2, -1}, 1001, 1);
+        String selectNested = "SELECT _string, _varbinary, _bigint, _struct._field FROM %s WHERE _struct._field = 1";
+        assertThat(onTrino().executeQuery(format(selectNested, trinoTableName)))
+                .containsOnly(row3);
+        assertThat(onSpark().executeQuery(format(selectNested, sparkTableName)))
+                .containsOnly(row3);
     }
 
     @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS, ICEBERG_REST, ICEBERG_JDBC}, dataProvider = "storageFormatsWithSpecVersion")
     public void testTrinoReadsSparkPartitionedTable(StorageFormat storageFormat, int specVersion)
     {
+        if (storageFormat == StorageFormat.ORC) {
+            throw new SkipException("Open Spark issue https://github.com/apache/iceberg/issues/3139 to read ORC table with nested partition column.");
+        }
         String baseTableName = toLowerCase("test_trino_reads_spark_partitioned_table_" + storageFormat);
         String trinoTableName = trinoTableName(baseTableName);
         String sparkTableName = sparkTableName(baseTableName);
         onSpark().executeQuery("DROP TABLE IF EXISTS " + sparkTableName);
 
         onSpark().executeQuery(format(
-                "CREATE TABLE %s (_string STRING, _varbinary BINARY, _bigint BIGINT) USING ICEBERG PARTITIONED BY (_string, _varbinary) TBLPROPERTIES ('write.format.default'='%s', 'format-version' = %s)",
+                "CREATE TABLE %s (_string STRING, _varbinary BINARY, _bigint BIGINT, _struct STRUCT<_field:STRING>) USING ICEBERG PARTITIONED BY (_string, _varbinary, _struct._field) TBLPROPERTIES ('write.format.default'='%s', 'format-version' = %s)",
                 sparkTableName,
                 storageFormat,
                 specVersion));
-        onSpark().executeQuery(format("INSERT INTO %s VALUES ('a', X'0ff102f0feff', 1001), ('b', X'0ff102f0fefe', 1002), ('c', X'0ff102fdfeff', 1003)", sparkTableName));
-
-        Row row1 = row("a", new byte[] {15, -15, 2, -16, -2, -1}, 1001);
-        String select = "SELECT * FROM %s WHERE _string = 'a'";
+        onSpark().executeQuery(format("INSERT INTO %s VALUES ('a', X'0ff102f0feff', 1001, named_struct('_field', 'field1')), ('b', X'0ff102f0fefe', 1002, named_struct('_field', 'field2')), ('c', X'0ff102fdfeff', 1003, named_struct('_field', 'field3'))", sparkTableName));
+
+        Row row1 = row("a", new byte[] {15, -15, 2, -16, -2, -1}, 1001, "field1");
+        String select = "SELECT _string, _varbinary, _bigint, _struct._field FROM %s WHERE _string = 'a'";
+        // ORC: Job aborted due to stage failure: Task 0 in stage 40.0 failed 1 times, most recent failure: Lost task 0.0 in stage 40.0 (TID 70) (spark executor driver): java.lang.IndexOutOfBoundsException: Index 1 out of bounds for length 1
```

Comment:
Same as for Parquet.

Reply:
I could not find any existing issue for PARQUET issue. Where Saprk returns null for the partitioned nested field.